lib/ansible/plugins/filter/flatten.yml

```diff
@@ -7,8 +7,8 @@ DOCUMENTATION:
   positional: _input, levels, skip_nulls
   options:
     _input:
-      description: First dictionary to combine.
-      type: dict
+      description: List to flatten.
+      type: list
```

Comment:
elements should be a list also

Reply:
Not necessarily:
```yaml
----
- gather_facts: no
  hosts: localhost
  connection: local
  tasks:
  - debug:
      var: not_flat | flatten
    vars:
      not_flat:
        - foo
        - - bar
          - foobar
        - {'foo': bar}
```

```
PLAYBOOK: play.yml *************
1 plays in play.yml

PLAY [localhost] *****************
TASK [debug] *******************
task path: /home/max/ac-rennes/play.yml:6
ok: [localhost] => {
    "not_flat | flatten": [
        "foo",
        "bar",
        "foobar",
        {
            "foo": "bar"
        }
    ]
}
```
elements: any maybe ?

Reply:
https://github.com/ansible/ansible/blob/cae4f90b21bc40c88a00e712d28531ab0261f759/lib/ansible/plugins/filter/core.py#L468-L486

> if it is not a list it is coerced into a list of one element,

I'm not completely sure what you mean here, `append` coerce it's parameter into a list ?

Reply:
you can disregard, but i meant `"a"|flatten `